src/Aspire.Hosting/Dcp/DcpExecutor.cs

```diff
@@ -1206,6 +1188,8 @@ private async Task CreateContainerAsync(AppResource cr, ILogger resourceLogger,
             spec.Ports = BuildContainerPorts(cr);
         }
 
+        spec.VolumeMounts = BuildBindMounts(modelContainerResource);
```

Comment:
The idea is to be able to add container mount annotations during the `AftertEndpointAllocated`. The same way endpoints are built in `CreateContainerAsync`. Otherwise we can't create file mounts with deterministic names based on the file content (for persistent containers).

NB: In the case of MySql/PgSql the target ports are currently used, which are static. Resolution is done thought host name. This means that at least this part of the configuration in invariant across restarts.